addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiDeviceBaseHandler.java

```diff
@@ -104,14 +106,15 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void onItemUpdate(String sid, String command, JsonObject message) {
         if (getItemId() != null && getItemId().equals(sid)) {
+            updateThingStatus();
```

Comment:
The issue I was having was that the read_ack response was returned and processed BEFORE the sensor Thing status was set to online, causing the values to be ignored by the framework.  It seems sensible to me that if we have received a json message for a specific device then that device must be online, so this ensures that we update the Thing status before updating the channel state.